FocusTarget Typing Suggestion

version: `"focus-trap-react": "^8.7.1"`

I recently needed to use `focusTrapOptions.initialFocus`

> `initialFocus?: FocusTarget | false;`

I tried using `HTMLDivElement`, but I got an error. Looking at _global.d.ts_, `HTMLDivElement` extends `HTMLElement`, so the suggestion would be to add a generic type, eg:

```ts
export type FocusTarget<T extends HTMLElement> = T | SVGElement | string | { (): T | SVGElement };
```

so that all elements that are extended from `HTMLElement` be accepted.

If I make this change, I get TSC errors like this:
```
index.d.ts:83:20 - error TS2314: Generic type 'FocusTargetOrFalse' requires 1 type argument(s).

83     initialFocus?: FocusTargetOrFalse;
```

So the type for the `initialFocus` option would then have to change to:
```
initialFocus?: FocusTargetOrFalse<SOMETHING?>;
```

I'm not sure what to put there. Seems like I'd have to specify a type, and then we're right back where we started.

Come to think of it, I'm surprised the current definition of
```typescript
export type FocusTargetOrFalse = HTMLElement | SVGElement | string | false | (() => HTMLElement | SVGElement | false;
```
is causing you trouble when you try to give an `HTMLDivElement` to a property like `initialFocus`. As you said earlier, `HTMLDivElement` extends from `HTMLElement`, so there should be no type issue with giving this type of element to the option. `HTMLDivElement` is compatible with `HTMLElement` in terms of interfaces since `HTMLDivElement is a superset of `HTMLElement`.

So why is this a problem at all? 🤔

--- a/.changeset/blue-peas-matter.md
+++ b/.changeset/blue-peas-matter.md
@@ -1,0 +1,5 @@
+---
+'focus-trap-react': minor
+---
+
+Bump focus-trap dependency to v6.7.0

--- a/package.json
+++ b/package.json
@@ -91,7 +91,7 @@
     "typescript": "^4.4.3"
   },
   "dependencies": {
-    "focus-trap": "^6.6.1"
+    "focus-trap": "^6.7.0"
   },
   "peerDependencies": {
     "prop-types": "^15.7.2",

